window_doc.py

```diff
@@ -37,33 +38,30 @@ def represented_file_is_valid(doc: str) -> bool:
             return False
 
         if not os.path.exists(doc):
-            app.notify("Document doesn't exist", f"The current window's document doesn't seem to exist on disk:\n\n{doc}")
+            app.notify("Document doesn't exist",
+                       f"The current window's document doesn't seem to exist on disk:\n\n{doc}")
             return False
 
         return True
 
-    def open_current_doc(cmd: str = None):
+    def open_current_doc(cmd: str = None) -> Optional[subprocess.CompletedProcess]:
         """Opens the current document in the default open handler, or passing it to {cmd}"""
 
         doc = actions.user.file_manager_current_path_or_doc()
         if not actions.user.represented_file_is_valid(doc):
-            return
+            return None
 
-        if cmd:
-            return actions.user.system_command_nb(f"{cmd} \"{doc}\"")
-        else:
-            return actions.user.system_command_nb(f"open \"{doc}\"")
+        return subprocess.run([cmd if cmd else "open", doc])
```

Comment:
Surprised `ui.launch` doesn't do the right thing here when passed a path, but apparently it doesn't! Would hard-code the path (`/usr/bin/open`), otherwise it seems as good a solution as any right now.

The other option would be `webbrowser.open(<file URL>)`.

Reply:
Done. Maybe later we want to provide generic actions for opening/revealing that we can just call here.